### PR TITLE
ignition: split ignition-setup into base and user

### PR DESCRIPTION
**Pull request by ajeddeloh**
Split ignition-setup into two units/scripts. This allows other distros
to override how we fetch user configs without changing how we fetch the
base config.

Obsoletes https://github.com/coreos/ignition-dracut/pull/80

### CODE REVIEW

File: dracut/30ignition/ignition-setup-base.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        echo "File ${src} does not exist.. Skipping copy"
+    fi
+}
+
+destination=/usr/lib/ignition/
+mkdir -p $destination
+
+# We will support grabbing a platform specific base.ign config
+# from the initrd at /usr/lib/ignition/platform/${PLATFORM_ID}/base.ign
+copy_file_if_exists "/usr/lib/ignition/platform/${PLATFORM_ID}/base.ign" "${destination}/base.ign"
```
Comment (Conan-Kudo) [inline dracut/30ignition/ignition-setup-base.sh]: What is `$PLATFORM_ID` defined by?

**Comment by Conan-Kudo** [comment]
cc: @laenion

File: dracut/30ignition/ignition-setup-base.sh (hunk 1 above)
Comment (bgilbert) [reply dracut/30ignition/ignition-setup-base.sh]: `ignition-generator` writes it into `/run/ignition.env`, which is added to the environment by `ignition-setup-base.service`.

**Comment by laenion** [comment]
Looks good so far, just https://github.com/coreos/ignition-dracut/blob/8551bc7fd89a80200d94d009ad3d17131fc11dcf/systemd/ignition-firstboot-complete.service#L20 is still left.

File: dracut/30ignition/ignition-setup-base.service [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Ignition (setup)
```
Comment (laenion) [inline dracut/30ignition/ignition-setup-base.service]: A distinction between system and user setup would be good.

Comment (ajeddeloh) [reply dracut/30ignition/ignition-setup-base.service]: Good catch, fixed.

**Comment by ajeddeloh** [comment]
Yeah I think it's best to leave the boot-complete as is. There's not a good way to split it and it can easily be overridden.

**Comment by ajeddeloh** [comment]
Tested and confirmed working both with and without user configs on fcos.

**Comment by laenion** [comment]
Tests on openSUSE also successful, can be merged from my perspective.

**Comment by Conan-Kudo** [comment]
Merge it! 👍 👍

File: dracut/30ignition/ignition-setup-base.service [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Ignition (setup base config)
+DefaultDependencies=false
+Before=ignition-complete.target
+
+Requires=local-fs-pre.target
```
Comment (cgwalters) [inline dracut/30ignition/ignition-setup-base.service]: What's the point of this line?  It seems odd to have a `Requires=` here paired with `Before=`.

Comment (ajeddeloh) [reply dracut/30ignition/ignition-setup-base.service]: Hmm good point. Think is what we do on CL, so it obviously doesn't hurt, and that target will get pulled in anyway. @bgilbert you added it in bootengine in https://github.com/coreos/bootengine/pull/135, was there a reason there?

Comment (bgilbert) [reply dracut/30ignition/ignition-setup-base.service]: Looks like I copied it from `ignition-disks.service`, and the line originally dates back to https://github.com/coreos/coreos-overlay/commit/757e8014c2c0a9e35c2a445e76a343ac4414478d.

Comment (cgwalters) [reply dracut/30ignition/ignition-setup-base.service]: OK, let's just remove the `Requires=` and get this merged.

Comment (ajeddeloh) [reply dracut/30ignition/ignition-setup-base.service]: Fixed.

**Comment by cgwalters** [comment]
Ah, looks like this conflicted with the other PR.

**Comment by ajeddeloh** [comment]
Rebased

**Pull request merged**